src/clj/fluree/db/util/bytes.cljc

```diff
@@ -35,15 +35,9 @@
 
 
 (defn long->UTF8
-  [n]
-  (loop [result '()
-         n'     n]
-    (if (= 0 n')
-      #?(:clj (byte-array result)
-         :cljs (.from js/Int8Array result))
-      (let [b   (bit-and n' 0xFF)
-            n'' (bit-shift-right n' 8)]
-        (if (= b 0)
-          (recur result n'')
-          (recur (conj result b)
-                 n''))))))
+  [l]
+  (->> [56 48 40 32 24 16 8 0] ;; byte offsets
+       (map (fn [i] (bit-and (bit-shift-right l i) 0xFF)))
```

Comment:
This probably doesn't matter in practice, but I think we should keep the loop structure here instead of using `map` + `remove`. This code ideally should be as fast as possible because it's often called repeatedly when processing a query.

There's a lot of other things to optimize, and the functional implementation is still way faster than having to do an async query the way it was before, so I don't feel that strongly about changing it back.

Reply:
You raise a good point, I've rewritten it to be a loop. `criterium/quick-bench` pegs it as a little more than twice as fast now - ~300ns per long vs ~850ns with the seq version. I've also added a property test to make sure that it's working for inputs we didn't consider.